### PR TITLE
Update the pull-request-template file to show more detailed informati…

### LINKED ISSUE #697: Update pull-request-template.md file for more detailed information
Url: `https://github.com/knative/test-infra/blob/master/.github/pull-request-template.md`

**The current**: 
```
<!--
Request Prow to automatically lint any go code in this PR:

/lint
-->

Fixes #

```
**Updated to**:
```
<!--
Request Prow to automatically lint any go code in this PR:

/lint
-->

**What this PR does, why we need it**:
Describe the role of this pr

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
**Which issue(s) this PR fixes**:
Fixes #123456

**Special notice to reviewersr**:
This issue is used to update the pr template

**Has this pr changed for users**:
Updated the template for contributors to create pr.
```

### PR DESCRIPTION
**Pull request by xichengliudui**
…on about the pr template created by the user

Signed-off-by: xichengliudui <1693291525@qq.com>

Fixes #697

### CODE REVIEW

**Comment by xichengliudui** [comment]
Update the pull-request-template file to show more detailed information about the user-created pr template, which makes it easier to review the pr

**Comment by xichengliudui** [comment]
I am very sorry that I only created pr after such a long time. Sorry

File: .github/pull-request-template.md [hunk 1]
```diff
@@ -4,4 +4,18 @@ Request Prow to automatically lint any go code in this PR:
...
+**What this PR does, why we need it**:
+
+<!--
+*Automatically closes linked issue when PR is merged.
+Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
+-->
+**Which issue(s) this PR fixes**:
 Fixes #
+
+**Special notice to reviewersr**:
```
Comment (adrcunha) [inline .github/pull-request-template.md]:
```suggestion
**Special notes to reviewers**:
```

File: .github/pull-request-template.md [hunk 2]
```diff
@@ -4,4 +4,18 @@ Request Prow to automatically lint any go code in this PR:
...
+
+**Has this pr changed for users**:
```
Comment (adrcunha) [inline .github/pull-request-template.md]:
```suggestion
**User-visible changes in this PR**:
```

File: .github/pull-request-template.md [hunk 3]
```diff
@@ -4,4 +4,18 @@ Request Prow to automatically lint any go code in this PR:
...
+```
+
```
Comment (adrcunha) [inline .github/pull-request-template.md]:
```suggestion
```

File: .github/pull-request-template.md [hunk 4]
```diff
@@ -4,4 +4,18 @@ Request Prow to automatically lint any go code in this PR:
...
+```
```
Comment (adrcunha) [inline .github/pull-request-template.md]:
```suggestion
```

Comment (xichengliudui) [reply .github/pull-request-template.md]: I didn't delete this because it's markdown syntax, like this:
```
xxxxxxxxxxxxxxxxxxxxxxxxxxx
```

**Comment by adrcunha** [comment]
The build tests are failing because:

```
I0508 01:06:56.971] .github/pull-request-template.md:19: MD031 Fenced code blocks should be surrounded by blank lines
I0508 01:06:56.971] 
I0508 01:06:56.971] A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

Maybe it's not worth keeping them in the template?

File: .github/pull-request-template.md [hunk 5]
```diff
@@ -4,4 +4,17 @@ Request Prow to automatically lint any go code in this PR:
...
+*Automatically closes linked issue when PR is merged.
+Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
+-->
+**Which issue(s) this PR fixes**:
 Fixes #
+
+**Special notes to reviewers**:
+
+**User-visible changes in this PR**:
+```
```
Comment (nagexiucai) [inline .github/pull-request-template.md]: > MD031 Fenced code blocks should be surrounded by blank lines

here?

Comment (adrcunha) [reply .github/pull-request-template.md]: Yes

**Comment by xichengliudui** [comment]
I am very sorry. I have been away from knative for a while, and I have updated the pr. Sorry again

**Comment by xichengliudui** [comment]
/test pull-knative-test-infra-build-tests
--

**Comment by xichengliudui** [comment]
Same problem as before

**Comment by adrcunha** [comment]
> Same problem as before

Because the same pattern is still there. Since you squashed the commits I can't say what exactly you changed in the last one. Personally I don't think the fenced block helps, so you might consider simply removing it.

**Comment by xichengliudui** [comment]
updated

File: .github/pull-request-template.md (hunk 3 above)
Comment (adrcunha) [reply .github/pull-request-template.md]: Ignored

**Pull request merged**